bin/commonmark

```diff
@@ -77,11 +78,27 @@ if (format === 'html') {
 }
 
 if (files.length === 0) {
-   if (os.platform() === "win32") {
-     inps.push(fs.readFileSync(0, 'utf-8'));
-   } else {
-     inps.push(fs.readFileSync('/dev/tty', 'utf-8'));
-   }
+  if (process.stdin.isTTY) {
+    if (os.platform() === 'win32') {
+      // Windows does not have a PTY that Node relies on.
+      // Instead, we read lines explicitly, tracking ^Z for EOF.
+      (async () => {
+        const rl = readline.createInterface({ input: process.stdin });
+        let inp = '';
+        for await (const line of rl) {
+          if (line === '\x1a') { break; }
+          inp += line;
+        }
+        const doc = parser.parse(inp);
+        const rendered = renderer.render(doc);
+        if (!options.time) { process.stdout.write(rendered); }
```

Comment:
Does this belong here? Aren't we going to fall through to line 109 and expect `inps` to contain an array of strings?
I am clueless about how async stuff works in JS, but is there a way to do this synchronously to make the code more straightforward?

Reply:
It will fall through, but with a `files` array that is empty (a requirement of the top if condition), so the rest of the code prints nothing.

There is no built-in way to do things synchronously with readline.
Node.js encourages asynchronous operation as it was its main philosophical tenet.
The fs API is an exemption to the rule.

There are external libraries that I think can do that, such as `readline-sync`.
It seems to do [some gnarly background-process spawning](https://github.com/anseki/readline-sync/blob/master/lib/readline-sync.js#L194-L206).

We can also convert the whole file to asynchronous operation, which is a more significant patch.

Reply:
I see.  It would be nice to avoid duplicating the parsing and rendering code like this -- if it takes making the whole thing async, maybe that's the best approach.

Reply:
Well, maybe it's not worth the effort. I can just merge this.

Reply:
Actually, I think this is not good as it is.  You fall through and then end up parsing another empty document and printing the result of rendering it.  I'd want to avoid that.